src/hooks/useHasContractExpired.ts

```diff
@@ -0,0 +1,21 @@
+import { useActiveProvider } from './useActiveProvider'
+import { useContract } from './useContract'
+import { FOX_ETH_FARMING_ADDRESS } from 'lib/constants'
+import farmAbi from 'abis/farmingAbi.json'
+import { useEffect, useState } from 'react'
+
+export const useHasContractExpired = () => {
+  const [expired, setExpired] = useState(false)
+  const provider = useActiveProvider()
+  const farmingRewardsContract = useContract(provider, null, FOX_ETH_FARMING_ADDRESS, farmAbi)
```

Comment:
When we get the routes in with the farming address coming from the url we can then swap this out to pull contract from the url.